source/javascripts/asciinema-player.js

```diff
@@ -0,0 +1,1213 @@
+2;b.N=function(a){var b=y(a);a=z(a);var d=y(a);a=vd(a);return c(b,d,a)};b.A=c;return b}();e=function(a,e,l){switch(arguments.length){case 0:return d.call(this);case 1:return c.call(this,a);case 2:return b.call(this,a,e);default:var h=null;if(2<arguments.length){h=0;for(var k=Array(arguments.length-2);h<k.length;)k[h]=arguments[h+2],++h;h=new Jb(k,0,null)}return f.A(a,e,h)}throw Error("Invalid arity: "+(arguments.length-1));};e.L=2;e.N=f.N;e.B=d;e.h=c;e.c=b;e.A=f.A;return e}()}
+function Zf(a){return function(){function b(b){if(0<arguments.length)for(var c=0,e=Array(arguments.length-0);c<e.length;)e[c]=arguments[c+0],++c;return a}b.L=0;b.N=function(b){E(b);return a};b.A=function(){return a};return b}()}
+var $f=function $f(a){switch(arguments.length){case 0:return $f.B();case 1:return $f.h(arguments[0]);case 2:return $f.c(arguments[0],arguments[1]);case 3:return $f.l(arguments[0],arguments[1],arguments[2]);default:for(var c=[],d=arguments.length,e=0;;)if(e<d)c.push(arguments[e]),e+=1;else break;return $f.A(arguments[0],arguments[1],arguments[2],new Jb(c.slice(3),0,null))}};$f.B=function(){return Ve};$f.h=function(a){return a};
+$f.c=function(a,b){return function(){function c(c,d,e){c=b.l?b.l(c,d,e):b.call(null,c,d,e);return a.h?a.h(c):a.call(null,c)}function d(c,d){var e=b.c?b.c(c,d):b.call(null,c,d);return a.h?a.h(e):a.call(null,e)}function e(c){c=b.h?b.h(c):b.call(null,c);return a.h?a.h(c):a.call(null,c)}function f(){var c=b.B?b.B():b.call(null);return a.h?a.h(c):a.call(null,c)}var h=null,k=function(){function c(a,b,c,e){var f=null;if(3<arguments.length){f=0;for(var h=Array(arguments.length-3);f<h.length;)h[f]=arguments[f+
+3],++f;f=new Jb(h,0,null)}return d.call(this,a,b,c,f)}function d(c,d,e,f){c=Af(b,c,d,e,f);return a.h?a.h(c):a.call(null,c)}c.L=3;c.N=function(a){var b=y(a);a=z(a);var c=y(a);a=z(a);var e=y(a);a=vd(a);return d(b,c,e,a)};c.A=d;return c}();h=function(a,b,h,u){switch(arguments.length){case 0:return f.call(this);case 1:return e.call(this,a);case 2:return d.call(this,a,b);case 3:return c.call(this,a,b,h);default:var m=null;if(3<arguments.length){m=0;for(var p=Array(arguments.length-3);m<p.length;)p[m]=
+arguments[m+3],++m;m=new Jb(p,0,null)}return k.A(a,b,h,m)}throw Error("Invalid arity: "+(arguments.length-1));};h.L=3;h.N=k.N;h.B=f;h.h=e;h.c=d;h.l=c;h.A=k.A;return h}()};
+$f.l=function(a,b,c){return function(){function d(d,e,f){d=c.l?c.l(d,e,f):c.call(null,d,e,f);d=b.h?b.h(d):b.call(null,d);return a.h?a.h(d):a.call(null,d)}function e(d,e){var
```

Comment:
Whats the source of this, is there an unminified version, we'll need to verify it doesn't contain anything unexpected before merging...

Reply:
unminified version will look like this
```
goog.provide('asciinema.player.frames');
goog.require('cljs.core');
goog.require('cljs.core.constants');
goog.require('asciinema.player.util');
asciinema.player.frames.frame = (function asciinema$player$frames$frame(time,data){
return (new cljs.core.PersistentVector(null,2,(5),cljs.core.PersistentVector.EMPTY_NODE,[time,data],null));
});
asciinema.player.frames.update_frame_data = (function asciinema$player$frames$update_frame_data(f,p__36865){
var vec__36866 = p__36865;
var time = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36866,(0),null);
var data = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36866,(1),null);
return (new cljs.core.PersistentVector(null,2,(5),cljs.core.PersistentVector.EMPTY_NODE,[time,(f.cljs$core$IFn$_invoke$arity$1 ? f.cljs$core$IFn$_invoke$arity$1(data) : f.call(null,data))],null));
});
asciinema.player.frames.map_frame_data = (function asciinema$player$frames$map_frame_data(f,frames){
return cljs.core.map.cljs$core$IFn$_invoke$arity$2((function (p1__36869_SHARP_){
return asciinema.player.frames.update_frame_data(f,p1__36869_SHARP_);
}),frames);
});
/**
 * Returns lazy seq of interleaved frames coming from two collections.
 */
asciinema.player.frames.interleave_frames = (function asciinema$player$frames$interleave_frames(frames1,frames2){
return (new cljs.core.LazySeq(null,(function (){
if(cljs.core.seq(frames1)){
if(cljs.core.seq(frames2)){
var vec__36870 = cljs.core.first(frames1);
var t1 = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36870,(0),null);
var _ = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36870,(1),null);
var f1 = vec__36870;
var vec__36873 = cljs.core.first(frames2);
var t2 = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36873,(0),null);
var ___$1 = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36873,(1),null);
var f2 = vec__36873;
if((t1 < t2)){
return cljs.core.cons(f1,(function (){var G__36876 = cljs.core.rest(frames1);
var G__36877 = frames2;
return (asciinema.player.frames.interleave_frames.cljs$core$IFn$_invoke$arity$2 ? asciinema.player.frames.interleave_frames.cljs$core$IFn$_invoke$arity$2(G__36876,G__36877) : asciinema.player.frames.interleave_frames.call(null,G__36876,G__36877));
})());
} else {
return cljs.core.cons(f2,(function (){var G__36878 = frames1;
var G__36879 = cljs.core.rest(frames2);
return (asciinema.player.frames.interleave_frames.cljs$core$IFn$_invoke$arity$2 ? asciinema.player.frames.interleave_frames.cljs$core$IFn$_invoke$arity$2(G__36878,G__36879) : asciinema.player.frames.interleave_frames.call(null,G__36878,G__36879));
})());
}
} else {
return frames1;
}
} else {
return null;
}
}),null,null));
});
/**
 * Adjust frame time by given offset.
 */
asciinema.player.frames.translate_frame = (function asciinema$player$frames$translate_frame(offset,p__36880){
var vec__36881 = p__36880;
var time = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36881,(0),null);
var data = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36881,(1),null);
return (new cljs.core.PersistentVector(null,2,(5),cljs.core.PersistentVector.EMPTY_NODE,[(time + offset),data],null));
});
/**
 * Adjust frame time to match given speed.
 */
asciinema.player.frames.accelerate_frame = (function asciinema$player$frames$accelerate_frame(speed,p__36884){
var vec__36885 = p__36884;
var time = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36885,(0),null);
var data = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36885,(1),null);
return (new cljs.core.PersistentVector(null,2,(5),cljs.core.PersistentVector.EMPTY_NODE,[(time / speed),data],null));
});
asciinema.player.frames.accelerate_xf = (function asciinema$player$frames$accelerate_xf(speed){
return cljs.core.map.cljs$core$IFn$_invoke$arity$1(cljs.core.partial.cljs$core$IFn$_invoke$arity$2(asciinema.player.frames.accelerate_frame,speed));
});
asciinema.player.frames.cap_relative_time_xf = (function asciinema$player$frames$cap_relative_time_xf(time_limit){
return cljs.core.map.cljs$core$IFn$_invoke$arity$1((function (p__36888){
var vec__36889 = p__36888;
var t = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36889,(0),null);
var data = cljs.core.nth.cljs$core$IFn$_invoke$arity$3(vec__36889,(1),null);
var frame = vec__36889;
if(cljs.core.truth_(time_limit)){
return new cljs.core.PersistentVector(null, 2, 5, cljs.core.PersistentVector.EMPTY_NODE, [(function (){var x__31135__auto__ = t;
var y__31136__auto__ = time_limit;
return ((x__31135__auto__ < y__31136__auto__) ? x__31135__auto__ : y__31136__auto__);
})(),data], null);
} else {
return frame;
}
}));
});

```

It's just a small part of code.

Reply:
@JonRowe do you have any suggestions about what to do?

Reply:
Ok, whats the source of the minimised version? Basically is there somewhere for ascii cinema I can download this from to verify? Sorry to be a pain, its just its a large opaque block of js going on our website...

Reply:
It's from here https://github.com/asciinema/asciinema-player/tree/v2.6.1
I would like to build it within rspec.github.io project but it has a lot of dependencies and quite painful to set up the infrastructure to build it